tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_axis_view.ts

```diff
@@ -96,16 +139,35 @@ export class LineChartAxisComponent {
   @Input()
   domDim!: Dimension;
 
-  getTicks(): number[] {
-    const domSize = this.axis === 'x' ? this.domDim.width : this.domDim.height;
-    return this.scale.ticks(
-      this.axisExtent,
-      getDomSizeInformedTickCount(domSize, this.gridCount)
+  @Input()
+  customFormatter?: Formatter;
+
+  /**
+   * Returns true if the major ticks must be shown.
+   *
+   * Major tick is required for temporal axis since tick string do not have sufficient
+   * information alone and is hard to fit time information without a major ticks.
+   */
+  shouldShowMajorTicks(): boolean {
+    const majorTickCount = this.getMajorTicks().length;
+    const minorTickCount = this.getTicks().length;
+
+    return (
+      this.scale instanceof TemporalScale &&
+      this.axisExtent[1] - this.axisExtent[0] < DAY_IN_MS &&
+      minorTickCount > majorTickCount &&
+      this.getMajorTicks().length <= 2
     );
   }
 
-  getTickString(tick: number): string {
-    return numberFormatter.formatTick(tick);
+  getMajorTicks() {
```

Comment:
The `: number[]` return type annotation on L167 is nice for me, who often forgets d3 function signatures.  Can we add it here, too?